spec/models/poll_spec.rb

```diff
@@ -93,21 +111,187 @@
+      poll.ready!
+      poll.update_attribute(:expire_at, 1.year.ago)
+
+      expect { poll.reload }.to change { poll.status }.to('finished')
+    end
+  end
+
+  context 'comments' do
+    it 'should have comments' do
+      expect(poll.comments).to eq([])
+    end
+  end
+
+  context 'check availible transitions' do
+    context 'when poll draft' do
+      it 'have status draft' do
+        expect(poll).to be_draft
+      end
+
+      it 'able to be ready' do
+        expect(poll.able_to_ready?).to be_truthy
+      end
+
+      it 'able to be deleted' do
+        expect(poll.able_to_delete?).to be_truthy
+      end
+
+      it 'not able to be finished' do
+        expect(poll.able_to_finish?).to be_falsey
+      end
+    end
+
+    context 'when poll ready' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'ready' }
+
+      it 'have status ready' do
+        expect(poll).to be_ready
+      end
+
+      it 'able to be draft' do
+        expect(poll.able_to_draft?).to be_truthy
+      end
+
+      it 'able to be deleted' do
+        expect(poll.able_to_delete?).to be_truthy
+      end
+
+      it 'able to be finished' do
+        expect(poll.able_to_finish?).to be_truthy
+      end
+    end
+
+    context 'when poll finished' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'finished' }
+
+      it 'have status finished' do
+        expect(poll).to be_finished
+      end
+
+      it 'able to be draft' do
+        expect(poll.able_to_draft?).to be_truthy
+      end
+
+      it 'able to be deleted' do
+        expect(poll.able_to_delete?).to be_truthy
+      end
+
+      it 'not able to be ready' do
+        expect(poll.able_to_ready?).to be_falsey
+      end
+    end
+
+    context 'when poll deleted' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'deleted' }
+
+      it 'have status deleted' do
+        expect(poll).to be_deleted
+      end
+
+      it 'able to be draft' do
+        expect(poll.able_to_draft?).to be_truthy
+      end
+
+      it 'not able to be ready' do
+        expect(poll.able_to_ready?).to be_falsey
+      end
+
+      it 'not able to be finished' do
+        expect(poll.able_to_finish?).to be_falsey
+      end
+    end
   end
 
-  it 'shold have error when reopen outdated' do
-    poll.update_attribute(:expire_at, 1.year.ago)
-    poll.closed!
+  context 'Check changing status callbacks.' do
+    context 'draft -> ready' do
+      let(:poll) { FactoryGirl.create :poll, :with_title }
+
+      it 'stay draft if poll have no options' do
+        poll.ready!
+
+        expect(poll).to be_draft
+      end
+
+      it 'have an error if poll have no options' do
+        expect { poll.ready! }.to change { poll.errors.count }.by(1)
+      end
+    end
+
+    context 'ready -> draft' do
+      let(:poll) { FactoryGirl.create :valid_poll, status: 'ready' }
+      let(:poll2) { FactoryGirl.create :valid_poll, status: 'ready' }
+      before(:each) { poll.vote!(user, [0, 1, 2]) }
```

Comment:
I try to do so and fail, such trait affect user factory and when I try to create users, who votes I get error 'username alreasy have been taken, email already have been taken'. Looks like sequence in user factory not re-evaluate and I not figure out why and what do do with it.